docs/content/5.integrations/2.guide/2.adding-headers-to-request.md

```diff
@@ -0,0 +1,284 @@
+# Pass Custom Headers in Requests
+
+In this tutorial, we'll demonstrate how to set up an interceptor to pass headers when making requests using the axios client inside the Vue Storefront SDK. This is useful when you need to append custom headers or cookies to your API calls.
+
+After passing the headers to the axios client, we'll also show you how to pass them to external services using the API Client.
+
+::danger
+Do not use createSharedComposable if creating a shared composable that uses the interceptor. This will cause the interceptor to only be created once, which will result in the headers being shared across instances.
+::
+
+## Prerequisites:
+Ensure you have the Vue Storefront SDK set up in your project.
+
+## Notes
+::info
+This guide uses the integrations boilerplate with the nuxt option. The same basic principles apply to other frameworks, but the implementation may vary.
+::
+
+All SDK modules export a client instance of axios. This is the axios instance that is used to make requests. You can access this instance by importing the `client` from the SDK.
+
+## Steps
+::steps
+#step-1
+**Import Required Modules**
+
+```javascript
+import { initSDK, buildModule } from '@vue-storefront/sdk';
+import { client, boilerplateModule, BoilerplateModuleType } from '../../packages/sdk/src';
+import useHeaders from 'path-to-useHeaders-file'; // Replace with the correct path to the `useHeaders` composable.
+```
+
+#step-2
+**Define the SDK Configuration**
+```javascript
+const sdkConfig = {
+  boilerplate: buildModule<BoilerplateModuleType>(boilerplateModule, {
+    apiUrl: 'http://localhost:8181/boilerplate',
+  }),
+};
+```
+#step-3
+**Fetch Headers with the useHeaders Composable**
+
+Use the `useHeaders` composable to fetch the necessary headers. This composable manages both server-side and client-side headers for you.
+
+```javascript
+const { addHeadersToState, headerData } = useHeaders();
+addHeadersToState();
+```
+
+#step-4
+**Setting Up Axios Interceptor**
+
+Before making requests using axios, establish an interceptor to automatically append the desired headers.
```

Comment:
```suggestion
Before making requests using axios, establish an interceptor to automatically append the desired headers.
```

Reply:
Did you intend to suggest a change?
<img width="809" alt="image" src="https://github.com/vuestorefront/vue-storefront/assets/45824492/898d7b91-68e7-41a5-8681-67642fbd8f75">